docs/notebooks/expected_improvement.pct.py

```diff
@@ -81,18 +79,13 @@ def build_model(data):
 #
 # The `optimize` method returns several things (see the `optimize` documentation for details), but we're only interested in the data, which captures the points where the objective was queried and the resulting values. Note that the optimizer updates the model in place.
 #
-# However, since the optimization loop catches errors so as not to lose progress, we must check if any errors occurred so we know the data is valid. We'll do that crudely here by re-raising any such errors. You may wish instead to use the history to restore the process from an earlier point.
+# However, the optimization loop catches errors so as not to lose progress, which means the optimization loop might not complete and the data from the last step may not exist. Here we'll handle this crudely by asking for the data regardless, which will re-raise the error if one did occur. For a review of how to handle errors systematically, there is a [dedicated tutorial](recovering_from_errors.ipynb).
 
 # %%
-from trieste.bayesian_optimizer import OptimizationResult
-
 bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
 
-result: OptimizationResult = bo.optimize(15, initial_data, model)
-
-if result.error is not None: raise result.error
-
-dataset = result.datasets[OBJECTIVE]
+result = bo.optimize(15, initial_data, model)
+dataset = result.try_final_data()[OBJECTIVE]
```

Comment:
I'd find it helpful to get at least a brief comment on what the `[OBJECTIVE]` does (can be a link to another place in the docs), as well as what `try_final_data` does (get you the data, or raise exception) - this is mostly explained in the markdown cell above, but connecting the textual description to which line in the code does that would be helpful (i.e., bo.optimize() will never throw an exception [perhaps with the exception of actual bugs in the code], but try_final_data() might - this is not clear from this notebook)

Reply:
i've updated the text. Is it clear now?